subsys/bluetooth/host/scan.c

```diff
@@ -524,6 +538,36 @@ void bt_hci_le_adv_ext_report(struct net_buf *buf)
 
 		evt = net_buf_pull_mem(buf, sizeof(*evt));
 
+		if (!ext_scan_buf) {
+			/* The previous report was truncated in the host.
+			 * Discard all remaining reports for that advertisement.
+			 */
+
+			if (!(evt->evt_type & BIT(5))) {
+				/* This event is the last event to be discarded.
+				 */
+				ext_scan_buf = net_buf_alloc(&ext_scan_buf_pool,
+							     K_NO_WAIT);
+				__ASSERT_NO_MSG(ext_scan_buf);
+			}
+
+			continue;
+		}
+
+		if (evt->length + ext_scan_buf->len > net_buf_max_len(ext_scan_buf)) {
+			size_to_copy = net_buf_max_len(ext_scan_buf) - evt->length;
+			truncate = true;
+		} else {
```

Comment:
Correct me if I am wrong, but if the size of the buffer is e.g. 1000, and we have already copied 900 bytes and this event has a size of 200, wouldn't `size_to_copy` become 800 rather than 100 with the current code? 

Shouldn't it be something like 
```suggestion
		if (evt->length + ext_scan_buf->len > net_buf_max_len(ext_scan_buf)) {
			size_to_copy = net_buf_max_len(ext_scan_buf) - ext_scan_buf->len;
			truncate = true;
		} else {
```